### PR TITLE
PRs: Move to use a temporary repo for pushing new prs

### PR DESCRIPTION
**Pull request by zeripath**
Whilst testing on my local machine I was able to consistently make a failure with making a PR from one particular repository to the same repository. I'm not certain why this is the case, but changing the pull structure to do the push from a temporary repository fixes this.

### CODE REVIEW

File: services/pull/pull.go [hunk 1]
```diff
@@ -311,23 +311,39 @@ func checkIfPRContentChanged(pr *models.PullRequest, oldCommitID, newCommitID st
 func PushToBaseRepo(pr *models.PullRequest) (err error) {
 	log.Trace("PushToBaseRepo[%d]: pushing commits to base repo '%s'", pr.BaseRepoID, pr.GetGitRefName())
 
+	// Clone base repo.
+	tmpBasePath, err := models.CreateTemporaryPath("pull")
```
Comment (6543) [inline services/pull/pull.go:315]: if func `PushToBaseRepo` is called in paralel - they still result in different tmpRepos ?
@zeripath

Comment (zeripath) [reply services/pull/pull.go:315]: Yup they do. Take a look at the code for create temporary path.

**Review by 6543 — APPROVED** [review]

**Review by mrsdizzie — APPROVED** [review]

File: services/pull/pull.go [hunk 2]
```diff
@@ -311,23 +311,39 @@ func checkIfPRContentChanged(pr *models.PullRequest, oldCommitID, newCommitID st
...
+	if err != nil {
+		log.Error("CreateTemporaryPath: %v", err)
+		return err
+	}
+	defer func() {
+		err := models.RemoveTemporaryPath(tmpBasePath)
+		if err != nil {
+			log.Error("Error whilst removing temporary path: %v", err)
```
Comment (guillep2k) [inline services/pull/pull.go]: It would be nice to log the path to remove, so the admin can remove it themselves.

Comment (zeripath) [reply services/pull/pull.go]:
```suggestion
			log.Error("Error whilst removing temporary path: %s Error: %v", tmpBasePath, err)
```

Comment (zeripath) [reply services/pull/pull.go]: It should literally never happen though.

**Pull request merged**